include/Dialect/Secret/IR/SecretOps.td

```diff
@@ -0,0 +1,161 @@
+#ifndef HEIR_INCLUDE_DIALECT_SECRET_IR_SECRETOPS_TD_
+#define HEIR_INCLUDE_DIALECT_SECRET_IR_SECRETOPS_TD_
+
+include "SecretDialect.td"
+include "SecretTypes.td"
+include "mlir/IR/BuiltinAttributeInterfaces.td"
+include "mlir/Interfaces/ControlFlowInterfaces.td"
+include "mlir/Interfaces/InferTypeOpInterface.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+
+class Secret_Op<string mnemonic, list<Trait> traits = []> :
+        Op<Secret_Dialect, mnemonic, traits> {
+  let cppNamespace = "::mlir::heir::secret";
+}
+
+def Secret_ConcealOp : Secret_Op<"conceal", [Pure]> {
+  let summary = "Convert a non-secret value into a secret";
+  let description = [{
+    Convert a value to a secret containing the same value.
+
+    This op represents a scheme-agnostic encryption operation, as well as a
+    "trivial encryption" operation which is needed for some FHE schemes. This
+    op is also useful for type materialization in the dialect conversion
```

Comment:
Should we file a follow-up issue for this for discussion to resolve when it feels like we know? Because in terms of lowering a noiseless vs true encryption will end up different - but maybe the case is that a lowering would be able to understand which is which depending on context? (if a secret key is not available it must be a trivial encryption)

Reply:
I think this dialect will probably change significantly before we can do that, e.g., by adding information about which secret key something is encrypted wrt, or equivalently a client/server distinction as to what is private to whom. Let's move that to https://github.com/google/heir/discussions/96 maybe.